shared/router-v2/shim.native.tsx

```diff
@@ -12,18 +12,26 @@ export const shim = (routes: any, isModal: boolean, isLoggedOut: boolean) =>
 
 export const getOptions = Shared.getOptions
 
-const edges = ['right', 'bottom', 'left'] as const
-
 const shimNewRoute = (Original: any, isModal: boolean, isLoggedOut: boolean, getOptions: any) => {
   // Wrap everything in a keyboard avoiding view (maybe this is opt in/out?)
   const ShimmedNew = React.memo(function ShimmedNew(props: any) {
     const navigationOptions =
       typeof getOptions === 'function'
         ? getOptions({navigation: props.navigation, route: props.route})
         : getOptions
-    const original = <Original {...props} />
-    const body = original
-    let wrap = body
+
+    let wrap = <Original {...props} />
+
+    const wrapInSafe = navigationOptions?.needsSafe || isModal || isLoggedOut
+    if (wrapInSafe) {
+      wrap = (
+        <SafeAreaProvider>
```

Comment:
we add a provider here so the inner safe area get better offsets that can be inside the keyboard avoiding view